# Case study: Using stents to prevent strokes

## Import data

In [1]:
import pandas as pd

df = pd.read_csv("https://raw.githubusercontent.com/kirenz/modern-statistics/main/data/stent.csv")
df.head()

group outcome     time
0  treatment  stroke  30 days
1  treatment  stroke  30 days
2  treatment  stroke  30 days
3  treatment  stroke  30 days
4  treatment  stroke  30 days

## Create a crosstable

In [2]:
pd.crosstab(df.group, [df.time, df.outcome],  margins=True)

time       30 days        365 days         All
outcome   no event stroke no event stroke     
group                                         
control        214     13      199     28  454
treatment      191     33      179     45  448
All            405     46      378     73  902

## Guided practice

### Strokes in treatment group

Task A: Compute the proportion of patients in the treatment group who had a stroke by the end of their first year. 

1) Calculate **group of interest**: 

- Patients in treatment group with stroke after 365 days

In [3]:
treatment_stroke_365 = df[(df["group"] == "treatment") & 
                          (df["outcome"] == "stroke") & 
                          (df["time"] == "365 days")].count()

print("Patients in treatment group with stroke after 365 days:", 
       treatment_stroke_365["outcome"])

Patients in treatment group with stroke after 365 days: 45


2. Calculate **reference group**: 

- All patients in treatment group after 365 days

In [4]:
treatment_365 = df[(df["group"] == "treatment") & 
                   (df["time"] == "365 days")].count()

print("All patients in treatment group after 365 days:", 
       treatment_365["outcome"])

All patients in treatment group after 365 days: 224


3. Calculate **proportion**: 

- Proportion of patients in the treatment group who had a stroke by the end of their first year

In [5]:
proportion = (treatment_stroke_365["outcome"]/treatment_365["outcome"]) * 100

print("Proportion with stroke:", proportion.round(0), "%")

Proportion with stroke: 20.0 %


## Practice

### Strokes in control group

Task B: Compute the proportion of patients in the control group who had a stroke by the end of their first year. 

<font class="your_turn">
    <strong>Your Turn</strong>
</font>

1. Use the code examples from task A to compute the proportion. 
2. Compare the results of A and B. Are the results surprising?

In [13]:
control_stroke_365 = df[(df["group"] == "control") & 
                          (df["outcome"] == "stroke") & 
                          (df["time"] == "365 days")].count()

print("Patients in control group with stroke after 365 days:", 
       control_stroke_365["outcome"])

Patients in control group with stroke after 365 days: 28


In [14]:
control_365 = df[(df["group"] == "control") & 
                   (df["time"] == "365 days")].count()

print("All patients in control group after 365 days:", 
       control_365["outcome"])

All patients in control group after 365 days: 227


In [15]:
proportion = (control_stroke_365["outcome"]/control_365["outcome"]) * 100

print("Proportion with stroke:", proportion.round(0), "%")

Proportion with stroke: 12.0 %


12% sind richtig.
**Are the results suprising?**
Ergebnisse sind nicht überraschend, da Kontrollgruppe eine geringere Stroke% besitzt.
